node/actors/bft/src/lib.rs

```diff
@@ -71,56 +56,39 @@ impl Config {
                 return Ok(());
             }
         }
 
         let cfg = Arc::new(self);
-        let (leader, leader_send) = leader::StateMachine::new(ctx, cfg.clone(), pipe.send.clone());
+        let (proposer_sender, proposer_receiver) = sync::watch::channel(None);
         let (replica, replica_send) =
-            replica::StateMachine::start(ctx, cfg.clone(), pipe.send.clone()).await?;
+            chonky_bft::StateMachine::start(ctx, cfg.clone(), pipe.send.clone(), proposer_sender)
+                .await?;
 
         let res = scope::run!(ctx, |ctx, s| async {
-            let prepare_qc_recv = leader.prepare_qc.subscribe();
-
             s.spawn_bg(async { replica.run(ctx).await.wrap("replica.run()") });
-            s.spawn_bg(async { leader.run(ctx).await.wrap("leader.run()") });
             s.spawn_bg(async {
-                leader::StateMachine::run_proposer(ctx, &cfg, prepare_qc_recv, &pipe.send)
+                chonky_bft::proposer::run_proposer(ctx, cfg.clone(), pipe.send, proposer_receiver)
                     .await
                     .wrap("run_proposer()")
             });
 
             tracing::info!("Starting consensus actor {:?}", cfg.secret_key.public());
 
-            // This is the infinite loop where the consensus actually runs. The validator waits for either
-            // a message from the network or for a timeout, and processes each accordingly.
+            // This is the infinite loop where the consensus actually runs. The validator waits for
+            // a message from the network and processes it accordingly.
             loop {
                 async {
-                    let InputMessage::Network(req) = pipe
+                    let InputMessage::Network(msg) = pipe
                         .recv
                         .recv(ctx)
                         .instrument(tracing::info_span!("wait_for_message"))
                         .await?;
-                    use validator::ConsensusMsg as M;
-                    match &req.msg.msg {
-                        M::ReplicaPrepare(_) => {
-                            // This is a hacky way to do a clone. This is necessary since we don't want to derive
-                            // Clone for ConsensusReq. When we change to ChonkyBFT this will be removed anyway.
-                            let (ack, _) = oneshot::channel();
-                            let new_req = ConsensusReq {
-                                msg: req.msg.clone(),
-                                ack,
-                            };
 
-                            replica_send.send(new_req);
-                            leader_send.send(req);
-                        }
-                        M::ReplicaCommit(_) => leader_send.send(req),
-                        M::LeaderPrepare(_) | M::LeaderCommit(_) => replica_send.send(req),
-                    }
+                    replica_send.send(msg);
```

Comment:
now that we have just one message consumer, it doesn't make sense to do this extra message passing. Either inline the replica logic here, or pass the message receiver to replica process directly.

Reply:
There still needs to be some conversion since the input channel to the actor is a simple mpsc channel, while the input channel to the replica process is a prunable_mpsc.
A more "radical" solution would be just to get rid of the whole actor pipe situation. It was only created because I thought we would have many components, in reality we ended up with just two big components. So I can refactor that quickly.

Reply:
up to you, as you say, we will need to clean it up eventually.

Reply:
I'll do it in a separate PR.